### DISCUSSION TITLE
Device service for larger files

### DISCUSSION THREAD

**Discussion by divyashree123456** [edgexfoundry/community#38] [Q&A] [answered]
Hi all ,
we are working on an application which requires sending 16mb video file (encrypted )data using device-virtual-go [Jakata]to the subscriber end.while testing we are facing following issues some times :
1. Some times Application service is getting killed without any error message.
2. At Subscriber end ,i will get data for some two to three iterations later connection refused error.
3. some times I am facing error "failed to publish  event to message bus" in device-virtual-go service.
please let me know for any suggestions .Thanks

**Comment by cloudxxx8** [comment]
Please check the log on the message bus (Redis or MQTT broker) when you get the error "failed to publish event to message bus"

**Reply by divyashree123456** [reply]
Hi ,
I have observed the logs ,when i get "failed to publish event to message bus" in device service , some times I will get error "overcoming of output buffer limit" in redis logs. and i will get "discarding bad pubsub connection:EOF" error in application service.as attached below:
![redis](https://user-images.githubusercontent.com/120623642/210552305-f4684d75-ff8a-4ab7-a3cb-556f9c4300f6.png)

![app-service](https://user-images.githubusercontent.com/120623642/210552326-bee83c0b-dda0-4eca-8a17-bccf2a18d3eb.png)

**Comment by lenny-goodell** [comment, marked as answer]
@divyashree123456 , I recommend you try using MQTT for the EdgeX MessageBus. To do this you will need to generate a new compose file with the services configured to use MQTT rather than Redis Pub/Sub. 

Use the **Compose Builder** tool here to accomplish this with the command `make gen no-secty ds-virtual mqtt-bus`
https://github.com/edgexfoundry/edgex-compose/tree/jakarta/compose-builder
Note this link is to the **jakarta** branch

See the README here for more details on this tool:
https://github.com/edgexfoundry/edgex-compose/blob/jakarta/compose-builder/README.md
and here for docs section:
https://docs.edgexfoundry.org/2.1/getting-started/Ch-GettingStartedUsers/#generate-a-custom-docker-compose-file

**Reply by divyashree123456** [reply]
Hi @Lenny-intel ,
Thanks for your suggestion I will try implementing it using MQTT for Edgex message bus.
